examples/performance-debugging/performance-debugging.cpp

```diff
@@ -0,0 +1,500 @@
+            ostream << "\t" << entry << std::endl;
+        }
+        ostream << "];" << std::endl;
+    }
+
+private:
+    const gko::LinOp *matrix;
+    const vec<ValueType> *b;
+    mutable std::vector<ValueType> rec_res_norms;
+    mutable std::vector<ValueType> true_res_norms;
+};
+
+
+}  // namespace loggers
+
+
+namespace {
+
+
+// Print usage help
+void print_usage(const char *filename)
+{
+    std::cerr << "Usage: " << filename << " [executor] [matrix file]"
+              << std::endl;
+    std::cerr << "matrix file should be a file in matrix market format. "
+                 "The file data/A.mtx is provided as an example."
+              << std::endl;
+    std::exit(-1);
+}
+
+
+void print_vector(const gko::matrix::Dense<> *vec)
+{
+    auto elements_to_print = std::min(gko::size_type(10), vec->get_size()[0]);
+    std::cout << "[" << std::endl;
+    for (int i = 0; i < elements_to_print; ++i) {
+        std::cout << "\t" << vec->at(i) << std::endl;
+    }
+    std::cout << "];" << std::endl;
+}
+
+
+}  // namespace
+
+
+int main(int argc, char *argv[])
+{
+    // Parametrize the benchmark here
+    // Pick a value type
+    using vtype = double;
+    // Pick a matrix format
+    using mtx = gko::matrix::Csr<vtype>;
+    // Pick a solver
+    using solver = gko::solver::Cg<vtype>;
+    // Pick a preconditioner type
+    using preconditioner = gko::matrix::IdentityFactory<vtype>;
+    // Pick a residual norm reduction value
+    auto reduction_factor = 1e-8;
+    // Pick a maximum iteration count
+    auto max_iters = 2000u;
+    // Pick an output file name
+    auto of_name = "data.txt";
+
+
+    // Simple shortcut
+    using vec = gko::matrix::Dense<vtype>;
+
+    // Print version information
+    std::cout << gko::version_info::get() << std::endl;
+
+    // Figure out where to run the code
+    std::shared_ptr<gko::Executor> exec;
+    if (argc == 1 || std::string(argv[1]) == "reference") {
+        exec = gko::ReferenceExecutor::create();
+    } else if (argc > 1 && std::string(argv[1]) == "omp") {
+        exec = gko::OmpExecutor::create();
+    } else if (argc > 1 && std::string(argv[1]) == "cuda" &&
+               gko::CudaExecutor::get_num_devices() > 0) {
+        exec = gko::CudaExecutor::create(0, gko::OmpExecutor::create());
+    } else {
+        print_usage(argv[0]);
+    }
+
+    // Read the input matrix file directory
+    std::string input_mtx = "data/A.mtx";
+    if (argc == 3) {
+        input_mtx = std::string(argv[2]);
+    }
+
+    // Read data: A is read from disk
+    // Create a StorageLogger to track the size of A
+    auto storage_logger = std::make_shared<loggers::StorageLogger>(exec);
+    // Add the logger to the executor
+    exec->add_logger(storage_logger);
+    // Read the matrix A from file
+    auto A = gko::share(gko::read<mtx>(std::ifstream(input_mtx), exec));
+    // Remove the storage logger
+    exec->remove_logger(gko::lend(storage_logger));
```

Comment:
This is the first time (and actually almost the last) you use the ginkgo namespace `gko::` here.
I would prefer if you would also add `gko::` to all `lend`, `clone` and so on above since it makes clear where the function comes from, and it would be also consistent with the other examples.